src/client/ua_client.c

```diff
@@ -427,8 +459,12 @@ UA_Client_runAsync(UA_Client *client, UA_UInt16 timeout) {
     if (retvalPublish != UA_STATUSCODE_GOOD)
         return retvalPublish;
 #endif
+    UA_StatusCode retval = UA_Client_backgroundConnectivityCheck(client);
```

Comment:
BackgroundConnectivityCheck does not respect the `timeout` variable.
Instead it uses the *global* timeout settings of the client.

Since this is a synchronous service call, we might block for a long time.
Can we make this an asynchronous call?

In the long term, once we have an event loop in the client, we can time-trigger the connectivity check.